scripts/azure-price-fetch/azureClientWrapper.go

```diff
@@ -0,0 +1,177 @@
+		RegionMap: make(map[string]map[string]sdk.ResourceSKU),
+	}
+
+	pager := a.priceClient.NewListPager(&sdk.RetailPricesClientListOptions{
+		APIVersion:  to.StringPtr("2023-01-01-preview"),
+		Filter:      to.StringPtr(a.buildQueryFilter(locationList)),
+		MeterRegion: to.StringPtr(`'primary'`),
+	})
+
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to advance page: %w", err)
+		}
+		for _, v := range page.Items {
+			if _, ok := pm.RegionMap[v.ArmRegionName]; !ok {
+				pm.RegionMap[v.ArmRegionName] = make(map[string]sdk.ResourceSKU)
+			}
+			pm.RegionMap[v.ArmRegionName][v.ArmSkuName] = v
+		}
+	}
+
+	return &pm, nil
+}
+
+func (a *AzureClientWrapper) getResourceGroupsAndLocationsInSubscription(ctx context.Context) (map[string]string, []string, error) {
+	rgToLocationMap := make(map[string]string)
+	locationSet := make(map[string]bool)
+	uniqueLocationList := []string{}
+
+	pager := a.rgClient.NewListPager(nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to advance page: %w", err)
+		}
+
+		for _, v := range nextResult.Value {
+			rgToLocationMap[*v.Name] = *v.Location
+			locationSet[*v.Location] = true
+		}
+	}
+
+	for v := range locationSet {
+		uniqueLocationList = append(uniqueLocationList, v)
+	}
+	return rgToLocationMap, uniqueLocationList, nil
+}
+
+func (a *AzureClientWrapper) getVmInfoFromResourceGroup(ctx context.Context, rgName string) (map[string]VirtualMachineInfo, error) {
+	vmInfoMap := map[string]VirtualMachineInfo{}
+
+	pager := a.vmssClient.NewListPager(rgName, nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to advance page: %w", err)
+		}
+
+		for _, v := range nextResult.Value {
+			vmsInfo, err := a.getVmInfoFromVmss(ctx, rgName, *v.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			for vmName, vmInfo := range vmsInfo {
+				vmInfoMap[vmName] = vmInfo
+			}
+		}
+	}
+
+	return vmInfoMap, nil
+}
+
+func (a *AzureClientWrapper) getVmInfoFromVmss(ctx context.Context, rgName string, vmssName string) (map[string]VirtualMachineInfo, error) {
+	vmInfo := make(map[string]VirtualMachineInfo)
+
+	pager := a.vmssVmClient.NewListPager(rgName, vmssName, nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to advance page: %w", err)
+		}
+
+		for _, v := range nextResult.Value {
+			vmInfo[*v.Name] = VirtualMachineInfo{
+				Name:        *v.Name,
+				OwningVMSS:  vmssName,
+				MachineType: *v.SKU.Name,
+			}
+		}
+	}
+
+	return vmInfo, nil
+}
+
+func (a *AzureClientWrapper) getRegionalVmInformationFromRgVmss(ctx context.Context, rgMap map[string]string) (*VmMap, error) {
+	vmMap := VmMap{
+		RegionMap: make(map[string]map[string]VirtualMachineInfo),
+	}
+
+	for rg, location := range rgMap {
+		m, err := a.getVmInfoFromResourceGroup(ctx, rg)
```

Comment:
What does `m` represent in this case? Since it's not immediately obvious, consider renaming the variable to what it represents.

Reply:
updated to `virtualMachineInfoByRg`